timemachine/cpp/src/neighborlist.cu

```diff
@@ -104,29 +110,35 @@ std::vector<std::vector<int> > Neighborlist<RealType>::get_nblist_host(
 
     // assert(N==N_);
 
+    int h_rebuild[1] = {1};
+
     double *d_coords = gpuErrchkCudaMallocAndCopy(h_coords, N*3*sizeof(double));
-    double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3*3*sizeof(double));    
+    double *d_box = gpuErrchkCudaMallocAndCopy(h_box, 3*3*sizeof(double));
+    int *d_rebuild = gpuErrchkCudaMallocAndCopy(h_rebuild, 1*sizeof(int));
 
+    cudaStream_t stream = static_cast<cudaStream_t>(0);
     this->build_nblist_device(
         N,
         d_coords,
         d_box,
         cutoff,
-        static_cast<cudaStream_t>(0)
+        d_rebuild,
+        true, // force rebuild
+        stream
     );
 
-    cudaDeviceSynchronize();
     const int B = this->B(); //(N+32-1)/32;
 
     unsigned long long MAX_TILE_BUFFER = B*B;
     unsigned long long MAX_ATOM_BUFFER = B*B*32;
 
     unsigned int h_ixn_count;
-    gpuErrchk(cudaMemcpy(&h_ixn_count, d_ixn_count_, 1*sizeof(*d_ixn_count_), cudaMemcpyDeviceToHost));
     std::vector<int> h_ixn_tiles(MAX_TILE_BUFFER);
     std::vector<unsigned int> h_ixn_atoms(MAX_ATOM_BUFFER);
-    gpuErrchk(cudaMemcpy(&h_ixn_tiles[0], d_ixn_tiles_, MAX_TILE_BUFFER*sizeof(int), cudaMemcpyDeviceToHost));
-    gpuErrchk(cudaMemcpy(&h_ixn_atoms[0], d_ixn_atoms_, MAX_ATOM_BUFFER*sizeof(unsigned int), cudaMemcpyDeviceToHost));
+    gpuErrchk(cudaMemcpyAsync(&h_ixn_count, d_ixn_count_, 1*sizeof(*d_ixn_count_), cudaMemcpyDeviceToHost, stream));
```

Comment:
Again, don't think we need to care too much about Async versions in the _host code since the IO back to python will be terrible anyways.